<!DOCTYPE html>
<html>
    <head>
        <meta charset"utf-8">
        <title>Bioinformatics Education</title>
        <meta name="description" content="Galaxy based lessons in Bioinformatics"> 
	    <link rel="stylesheet" href="../styles.css"> 
    </head>
    
    <body>
        <header class="site-header">

        <div id="container">
            <p class="site-title">Bioinformatics Tutorials using Galaxy</p>
            <div id="menu">
                <ul id="nav">
                <li><a href="../index.html">Differential RNA Expression - HiSAT2</a></li>
                <li><a class="active" href="./rna-seq-star.html">Differential RNA Expression - STAR</a></li>
                <li><a href="../assembly/assembly.html">Genomic Assembly</a></li>
                </ul>
            </div><br>
       
            <div class="tutorial">
                <h2 id="step-no">1.</h2>
                <div class="step-descrip"> 
                    Making frequent use of the 'History' tab within Galaxy will enable you to accurately manage the workflow of whatever pipeline you are managing. To start off, we will first import the data necessary for this step. To do that, navigate to the following link: <a href="https://usegalaxy.org/u/niemasd/h/rna-seq-methods">https://usegalaxy.org/u/niemasd/h/rna-seq-methods</a> and click 'Import history'. Now let's name the imported history "Diff. RNA Expression w/ STAR".
                </div>
                <div class="step-img"> 
                    <img src="./step1.png" width=150> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">2.</h2>
                <div class="step-descrip"> 
                    Given all of these reads, it is a good idea to first organize the data in a way that makes sense to you. For me, that involves rewriting the names of each fastq file to something more meaningful to me. To do this, click on each file name and then click on the pencil symbol to edit their attributes. While not necessary, as the tutorial proceeds we will be making a bunch of files in Galaxy's history and it will be useful to have clearly labeled files. My renamed files appear here.
                </div>
                <div class="step-img"> 
                    <img src="./step2.png" width=150> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">3.</h2>
                <div class="step-descrip"> 
                    The first step is to align these reads to the human reference genome. For this example, we will make use of the hg19 reference genome. We will use STAR to align the reads. To make use of the tool, use the Search Tools box on the left hand panel to find STAR. Click on STAR to bring up the options for the tool.  Use the paired-end as a collection option and choose paired end reads. Finally, select the hg19 reference genome within the reference genome panel and hit execute. Do this for each collection of reads, and rename each file after STAR finishes running.
                </div>
                <div class="step-img"> 
                    <img src="./step3.png" width=300> </img>
                </div>
            </div>

            <div class="tutorial">
                <h2 id="step-no">4.</h2>
                <div class="step-descrip"> 
                    The next step is to use StringTie to assemble the alignments into potential transcripts. To do this, search for StringTie on the lefthand panel and select the tool. Next, change the input to a dataset collection and select one of the created BAM files. Hit execute. Repeat this process for each of the BAM files generated from Step 3.
                </div>
                <div class="step-img"> 
                    <img src="./step4.png" width=300> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">5.</h2>
                <div class="step-descrip"> 
                    After renaming the outputted files to something more descriptive, we have to generate a unified transcriptome assembly. Before we can do that step, we must merge all of the disparate assemblies into one dataset. To do that, navigate to the tool panel and search for 'Merge Collections.' Click on the tool and add each of the StringTie results to merge into a single collection list of datasets. 
                </div>
                <div class="step-img"> 
                    <img src="./step5.png" width=300> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">6.</h2>
                <div class="step-descrip"> 
                     We now need to retreive a copy of the genome annotation file for the hg19 genome assembly. This will be used by StringTie merge as a reference annotation file to quantify the number of transcripts. To do this, navigate to <a href="http://genome.ucsc.edu/cgi-bin/hgTables?command=start" target="_blank">http://genome.ucsc.edu/cgi-bin/hgTables?command=start</a> and choose the hg19 assembly. Send the output to Galaxy. 
                </div>
                <div class="step-img"> 
                    <img src="./step6.png" width=300> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">7.</h2>
                <div class="step-descrip"> 
                     We are finally ready to run StringTie merge. To do this, navigate to the lefthand tools panel and search for StringTie merge. Click on the tool, and select the output of Step 5 in the Transcripts input. Select the outputted UCSC GTF file as the Reference annotation. Hit execute with the rest of the options untouched. Rename the output as Merged Transcriptome.  
                </div>
                <div class="step-img"> 
                    <img src="./step7.png" width=300> </img>
                </div>
            </div>

            <div class="tutorial">
                <h2 id="step-no">8.</h2>
                <div class="step-descrip"> 
                     To quantify the number of transcripts in each sample, we will make use of 'Feature Counts', another tool available in Galaxy. Select as the alignment file, a collection of the STAR alignments. The Gene Annotation File should be the Merged Transcriptome (output of StringTie merge). Additionally, click on 'Options for paired-end reads.' For the option of 'Count fragments instead of reads,' change the value to 'Enabled, fragments (or templates) will be counted.' Execute this command. You will need to do this for each of the 4 alignment collections. 
                </div>
                <div class="step-img"> 
                    <img src="./step8.png" width=300> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">9.</h2>
                <div class="step-descrip"> 
                    The final step is to run DEseq2 to quantify the differential expression between the UHRR data and the Brain data. To do this, select the DEseq2 tool and input 4 factor levels. For each factor level label it from the set of samples (UHRR1, UHRR2, Brain1, and Brain2) and for each designate the appropriate featureCounts file. Additionally, select "Yes" for the options "Output Normalized Counts Table" and "Output all levels vs all levels of primary factor (use when you have >2 levels for primary factor)." Label the resultant files with the STAR pipeline used to generate them.                   
                </div>
                <div class="step-img"> 
                    <img src="./step9.png" width=300> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">10.</h2>
                <div class="step-descrip"> 
                    Now download and inspect the results from the plot file. Spend some time thinking about the heatmap on the 2nd page, then proceed to Step 11.
                </div>
                <div class="step-img"> 
                    <img src="./step10.png" width=150> </img>
                </div>
            </div>
            
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>


            <div class="tutorial">
                <h2 id="step-no">11.</h2>
                <div class="step-descrip"> 
                    Compare this heatmap to the heatmap produced from the HiSAT2 pipeline. What does this comparison tell you about the differences between STAR/HiSAT2? Go back to the log files produced by both alignment runs, what are the significant differences? Why do you think people use one tool over the other? You may find my run's plots here in case you ran into trouble: <a href="./plots.pdf">DEseq2 Results</a>.
                </div>
                <div class="step-img"> 
                    <img src="./step11.png" width=300> </img>
                </div>
            </div>
        </div>
    </body>
