<!DOCTYPE html>
<html>
    <head>
        <meta charset"utf-8">
        <title>Bioinformatics Education</title>
        <meta name="description" content="Galaxy based lessons in Bioinformatics"> 
	    <link rel="stylesheet" href="./styles.css"> 
    </head>
    
    <body>
        <header class="site-header">

        <div id="container">
            <p class="site-title">Bioinformatics Tutorials using Galaxy</p>
            <div id="menu">
                <ul id="nav">
                <li><a class="active" href="./index.html">Differential RNA Expression - HiSAT2</a></li>
                <li><a href="./star/rna-seq-star.html">Differential RNA Expression - STAR</a></li>
                <li><a href="./assembly/assembly.html">Genomic Assembly</a></li>
                <li><a href="./trio/trio.html">Learning from a Trio</a></li>
                <li><a href="./wgs-vs-wes/wgs-vs-wes.html">WGS vs WES</a></li>
                </ul>
            </div><br>
       
            <div class="tutorial">
                <h2 id="step-no">1.</h2>
                <div class="step-descrip"> 
                    This tutorial will walk you through the steps of performing differential expression analysis on Galaxy's platform using the HiSAT2-StringTie protocol described in <a href="https://www.nature.com/articles/nprot.2016.095" target="_blank">Pertea et al. 2016</a>. Due to the large memory requirements of each of the datasets. We have already performed the alignment step for you! Please navigate to the <a href="https://usegalaxy.org/library/list#folders/Ff7c0c3182168bf3a" target="_blank">following link</a> and click on 'RNA Sequencing', then 'HiSAT2-Aligned_Data'. Select all of the BAM files in this folder and copy them to a new history for this analysis. (We called ours RNA-Diff.Exp.Analysis_HiSAT2). <pre><a href="./hisat2/step1.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">2.</h2>
                <div class="step-descrip"> 
                    We are now going to organize the HiSAT2 results. The best way to do that in Galaxy is to make use of their own data structures. Specifically, a 'Dataset List.' The way to construct this is to navigate to the history, and clicking on the gray button with a checkbox near the top of the history. Then click on each of the two replicates from each sample and click on 'For all selected...' and then click on 'Build Dataset List.' Do this for each of the samples (UHRR1, UHRR2, BRAIN1, and BRAIN2). <pre><a href="./hisat2/step2.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">3.</h2>
                <div class="step-descrip"> 
                    The next step is to run StringTie to quantify the transcript levels after the mapped HiSAT2 results. To do this, navigate to the tool panel on the left and search "StringTie". Click on the first result and then change the 'Input mapped reads' to the Dataset collection option. Select one of the HiSAT2 alignment collections made in the previous step and click execute without changing any other options. Repeat for all other dataset collections. <pre><a href="./hisat2/step3.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
            

            <div class="tutorial">
                <h2 id="step-no">4.</h2>
                <div class="step-descrip"> 
                    When each of the StringTie jobs have finished running, the next step is to create a unified transcriptome assembly. This is because each of the previous runs was conducted without a reference transcriptome. To do this, we will make use of StringTie merge; however, we first need to group each of the StringTie results together. To do this navigate to the tool panel and search for 'Merge Collections.' Click on the tool and add each of the StringTie results to merge into a single collection list of datasets. <pre> <a href="./hisat2/step4.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
        
            <div class="tutorial">
                <h2 id="step-no">5.</h2>
                <div class="step-descrip"> 
                    We now need to use a copy of the reference genome annotation file for the hg19 genome assembly to quantify the number of transcripts in each sample. We have already downloaded and uploaded this file onto our public server. Add it to your history by navigating to the <a href="https://usegalaxy.org/library/list#folders/Ff7c0c3182168bf3a" target="_blank">initial link</a>, and navigate to 'RNA Sequencing' and 'Annotation'. Select the 'hg19 GTF' and export it to your current History. <pre> <a href="./hisat2/step5.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
        
            <div class="tutorial">
                <h2 id="step-no">6.</h2>
                <div class="step-descrip"> 
                   We are now ready to run StringTie merge and merge the various transcripts. To do this, navigate to the lefthand tools panel and search for StringTie merge. Click on the tool, and select the output of Step 4 in the Transcripts input. Select the copied GTF file (Step 5) as the Reference annotation. Hit execute with the rest of the options untouched. Rename the output as Merged Transcriptome. <pre> <a href="./hisat2/step6.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
        
            <div class="tutorial">
                <h2 id="step-no">7.</h2>
                <div class="step-descrip"> 
                    To quantify the number of transcripts in each sample, we will make use of 'featureCounts', another tool available in Galaxy. Select as the alignment file, one of the collections of the HiSAT2 alignments. The Gene Annotation File should be the Merged Transcriptome (output of Step 6). Additionally, click on 'Options for paired-end reads.' For the option of 'Count fragments instead of reads,' change the value to 'Enabled, fragments (or templates) will be counted.' Execute this command. You will need to do this for each of the 4 alignment collections.<pre><a href="./hisat2/step7.png" target="_blank">See image hint</a></pre>
                </div>
            </div>
        
            <div class="tutorial">
                <h2 id="step-no">10.</h2>
                <div class="step-descrip"> 
                    The final step is to run DEseq2 to quantify the differential expression between the UHRR data and the Brain data. To do this, select the DEseq2 tool and input 4 factor levels. For each factor level label it from the set of samples (UHRR1, UHRR2, Brain1, and Brain2) and for each designate the appropriate featureCounts file. Additionally, select "Yes" for the options "Output Normalized Counts Table" and "Output all levels vs all levels of primary factor (use when you have >2 levels for primary factor)." Label the resultant files with the HiSAT2 pipeline used to generate them. <pre><a href="./hisat2/step8.png" target="_blank">See image hint</a></pre>
                </div>
            </div>

            <div class="tutorial">
                <h2 id="step-no">11.</h2>
                <div class="step-descrip"> 
                    Now download and inspect the results. See if you can make sense of the table in 'DEseq2 results' and the normalized counts file. Checking out DEseq2's <a href="https://bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf">documentation</a> might be helpful. Then spend some time thinking about the heatmap on the 2nd page of the plots file. When you've considered this, proceed to Step 12.
                </div>
            </div>
            
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>
            <br>

            <div class="tutorial" >
                <h2 id="step-no">12.</h2>
                <div class="step-descrip"> 
                    What do the colors of the various boxes represent? Why is the diagonal always 0? What can be said about the 'distance' between the UHRR samples (Universal Human RNA) and the Brain samples? Why is the distance less replicates? What can be concluded about the mRNA content of brain tissue versus other human tissue? What are future experiments that can be done? What would you focus on next? You may find my run's plots here in case you ran into trouble: <a href="./hisat2/plots.pdf">DEseq2 Results</a>.
                </div>
                <div class="step-img"> 
                    <img src="./hisat2/step12.png" width=200> </img>
                </div>
            </div>
            
            <div class="tutorial">
                <h2 id="step-no">Questions!</h2>
                <div class="step-descrip"> 
                    <ol>
                        <li> Put questions here.
                    </ol>
                </div>
            </div>
        </div>
    </body>
